hermes-frontend/src/main/java/pl/allegro/tech/hermes/frontend/buffer/BackupMessagesLoader.java

```diff
@@ -68,6 +74,7 @@ public BackupMessagesLoader(BrokerMessageProducer brokerMessageProducer,
         this.messageMaxAgeHours = config.getIntProperty(MESSAGES_LOCAL_STORAGE_MAX_AGE_HOURS);
         this.resendSleep = config.getIntProperty(KAFKA_PRODUCER_ACK_TIMEOUT) + secondsToWaitForTopicsCache * 1000;
         this.maxResendRetries = config.getIntProperty(MESSAGES_LOCAL_STORAGE_MAX_RESEND_RETRIES);
+        this.executorService = executorService;
     }
 
     public void loadMessages(MessageRepository messageRepository) {
```

Comment:
we could create the thread pool and close it in finally in this method, not to allocate resources for the lifetime of Frontend instance